libcst/codemod/commands/apply_stub_annotations.py

```diff
@@ -0,0 +1,506 @@
+                    name = self._add_annotation_to_imports(item.slice.value)
+                    new_index = item.slice.with_changes(value=name)
+                    new_slice.append(item.with_changes(slice=new_index))
+                else:
+                    if isinstance(item.slice, cst.Index) and not isinstance(
+                        item.slice.value, cst.Name
+                    ):
+                        new_index = item.slice.with_changes(
+                            value=self._handle_Index(item.slice, item)
+                        )
+                        item = item.with_changes(slice=new_index, comma=None)
+                    new_slice.append(item)
+            return node.with_changes(slice=new_slice)
+        elif isinstance(slice, cst.Index):
+            return self._handle_Index(slice, node)
+        else:
+            return node
+
+    def _create_import_from_annotation(self, returns: cst.Annotation) -> cst.Annotation:
+        annotation = returns.annotation
+        if isinstance(annotation, cst.Attribute):
+            attr = self._add_annotation_to_imports(annotation)
+            return cst.Annotation(annotation=attr)
+        if isinstance(annotation, cst.Subscript):
+            value = annotation.value
+            if isinstance(value, cst.Name) and value.value == "Type":
+                return returns
+            return cst.Annotation(annotation=self._handle_Subscript(annotation))
+        else:
+            return returns
+
+    def _add_to_imports(
+        self, names: List[cst.ImportAlias], module: cst.BaseExpression, key: str
+    ) -> None:
+        names_as_string = [_get_name_as_string(name.name) for name in names]
+        set_names = set(names_as_string)
+        if key not in self.imports:
+            # pyre-fixme[6]: Expected `Union[Attribute, Name]` for 2nd param but got
+            #  `BaseExpression`.
+            self.imports[key] = ImportStatement(names=set_names, module=module)
+        else:
+            import_statement = self.imports[key]
+            for name in set_names:
+                if name not in import_statement.names:
+                    import_statement.names.add(name)
+
+    def _import_parameter_annotations(
+        self, parameters: cst.Parameters
+    ) -> cst.Parameters:
+        def update_annotations(parameters: Sequence[cst.Param]) -> List[cst.Param]:
+            updated_parameters = []
+            for parameter in list(parameters):
+                annotation = parameter.annotation
+                if annotation is not None:
+                    parameter = parameter.with_changes(
+                        annotation=self._create_import_from_annotation(annotation)
+                    )
+                updated_parameters.append(parameter)
+            return updated_parameters
+
+        return parameters.with_changes(params=update_annotations(parameters.params))
+
+
+class TypeTransformer(cst.CSTTransformer):
```

Comment:
Can we make this class a `ContextAwareTransformer` and have the `apply_stub_annotations` as static method?
This is to make it consistent to our other transformer helpers like `AddImportsVisitor` and `RemovedNodeVisitor ` to prevent future confusion.

```
class TypeTransformer(ContextAwareTransformer):
    """ Add detailed documentation. """
    @staticmethod
    def apply_stub_annotations(sub_module: cst.Module):
        # collect stub type annotations using TypeCollector and store the collected data.

```
We can use GatherImportsVisitor to collect imports in `visit_Module` like this:
https://github.com/Instagram/LibCST/blob/9f7018902925f6404f4df8a36045c81cb53424b8/libcst/codemod/visitors/_add_imports.py#L169-L173

Once you finish that, you can add to the library of transform document by updating  https://github.com/Instagram/LibCST/blob/0ff9a6c5b7f2380dfeca4be01644c9d7e9146a30/docs/source/codemods.rst#L142-L150